src/v/raft/consensus.cc

```diff
@@ -938,6 +938,42 @@ ss::future<std::error_code> consensus::replace_configuration(
       });
 }
 
+ss::future<std::error_code>
+consensus::revert_configuration_change(model::revision_id revision) {
+    auto u = co_await _op_lock.get_units();
+    auto latest_cfg = config();
+    // latest configuration is of joint type
+    if (latest_cfg.type() == configuration_type::simple) {
+        vlog(_ctxlog.info, "no configuration changes in progress");
+        co_return errc::invalid_configuration_update;
+    }
+
+    if (latest_cfg.revision_id() > revision) {
+        co_return errc::invalid_configuration_update;
+    }
+    latest_cfg.revert_configuration_change();
+    co_return co_await replicate_configuration(
+      std::move(u), std::move(latest_cfg));
+}
+
+ss::future<std::error_code>
+consensus::abort_configuration_change(model::revision_id revision) {
```

Comment:
These functions are identical apart from one line, right?  Maybe collapse into a helper that takes that one line as a lambda.

Reply:
good point